textpipe/doc.py

```diff
@@ -74,23 +91,28 @@ def detect_language(self, hint_language=None):
                                          bestEffort=True)
         return best_guesses[0][1]
 
-    @property
```

Comment:
I don't think it makes sense to make `self.spacy_doc()` a property; in my opinion, it should should be private to the Doc class, and only be used to compute properties of Doc objects that are exposed. So to me, it would make more sense to change `self.spacy_doc()` to `self._spacy_doc()`

Reply:
Changed `self.spacy_doc()` to `self._spacy_doc()`. Curious what you think about this @dodijk.

Reply:
I don't really see why we would want to keep the `self.spacy_doc` private. If someone comes up with something fun to do with that, go ahead. We can't be reimplementing all the functionality from SpaCy's `Doc` into `textpipe.doc`, right? 

Getting this `spacy_doc` seems more like property than a method to me, that is why I proposed to call this action `spacy_nlp` and leave the `spacy_doc` property for the output of the plain vanilla call to this `spacy_nlp` method.

Reply:
I think that it would make more sense to actually wrap everything you want to use in Textpipe's Doc than to use Spacy's Doc directly. Textpipe's README says that it is `A uniform interface with thin wrappers around state-of-the-art NLP packages.` In my opinion, exposing the spacy_doc directly does not match this definition.

Reply:
Alright, I am fine with making the `spacy_doc` private. I do still believe it should remain a property, with `_spacy_nlp` as a (private) method.

Reply:
I think there should be an interface to the internals of the doc for a) debugging b) manipulation. It makes sense for it to be a property since it is precalculated. I am not particularly inclined towards a particular method name for this

Reply:
+1 on thinly wrapping/making spacy_doc private